src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/TokenBucketRateLimiter.cs

```diff
@@ -380,7 +380,29 @@ public RequestRegistration(int tokenCount, TaskCompletionSource<RateLimitLease>
             public TaskCompletionSource<RateLimitLease> Tcs { get; }
 
             public CancellationTokenRegistration CancellationTokenRegistration { get; }
+        }
 
+        private class WrappedTCS : TaskCompletionSource<RateLimitLease>
```

Comment:
Can this type be moved to the base class? So it's doubled from `ConcurrencyLimiter` (almost).

Reply:
There is a lot of shared code between TokenBucket and ConcurrencyLimiter, planning on sharing a lot of it in a future change.